packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -1311,6 +1313,26 @@ abstract class RenderSliver extends RenderObject {
   @override
   void handleEvent(PointerEvent event, SliverHitTestEntry entry) { }
 
+  /// Whether the [SemanticsNode]s associated with this [RenderSliver] should
+  /// be excluded from the semantic scrolling area.
+  ///
+  /// [RenderSliver]s that stay on the screen even though the user has scrolled
+  /// past them (e.g. a pinned app bar) should set this to `true`.
+  set excludeFromSemanticsScrolling(bool value) {
+    if (_excludeFromSemanticsScrolling == value)
+      return;
+    _excludeFromSemanticsScrolling = value;
+    markNeedsSemanticsUpdate();
+  }
+  bool _excludeFromSemanticsScrolling = false;
+
+  @override
+  SemanticsAnnotator get semanticsAnnotator => _excludeFromSemanticsScrolling ? _annotate : null;
```

Comment:
i would consider moving this just to the sliver_persistent_header.dart rather than having any of this logic in the root sliver class